events/views.py

```diff
@@ -1,11 +1,48 @@
 from django.shortcuts import render
 from django.http import HttpResponse
+from django.shortcuts import render, redirect
+from django.views.generic.edit import CreateView
 
-from .forms import EventPlaceForm, EventCategoryForm
+
+from .forms import EventForm, EventFormSet, EventPlaceForm, EventCategoryForm
+
+
+class EventCreateView(CreateView):
+    form_class = EventForm
+    template_name = "events/create_event.html"
+
+    def get_context_data(self, **kwargs):
+        context = super(EventCreateView, self).get_context_data(**kwargs)
+        context["event_formset"] = EventFormSet()
+        return context
+
+    def post(self, request, *args, **kwargs):
+        self.object = None
+        form = self.get_form()
+        event_formset = EventFormSet(
+            self.request.POST, self.request.FILES, instance=form.instance
+        )
+        print(event_formset.is_valid(), "is this true")
+        if form.is_valid() and event_formset.is_valid():
+            return self.form_valid(form, event_formset)
+        else:
+            return self.form_invalid(form, event_formset)
+
+    def form_valid(self, form, event_formset):
+        self.object = form.save(commit=False)
+        self.object.user = self.request.user
+        self.object.save()
```

Comment:
Why self.object is changed here. Don't create instance variables in local scope.